types/jsonwebtoken/index.d.ts

```diff
@@ -206,5 +227,5 @@ export function verify(
  * [options] - Options for decoding
  * returns - The decoded Token
  */
-export function decode(token: string, options: DecodeOptions & { json: true } | DecodeOptions & { complete: true }): null | { [key: string]: any };
-export function decode(token: string, options?: DecodeOptions): null | { [key: string]: any } | string;
+export function decode(token: string, options: DecodeOptions & { json: true } | DecodeOptions & { complete: true }): null | Jwt;
+export function decode(token: string, options?: DecodeOptions): null | JwtPayload | string;
```

Comment:
Perhaps an optional type argument could be considered too?
```suggestion
export function decode<T = Jwt>(token: string, options: DecodeOptions & { json: true } | DecodeOpti
export function decode<T = JwtPayload>(token: string, options?: DecodeOptions): null | T | string;
```

Reply:
or `extends` rather than `=`
but anyhow that would violate https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md :(

Reply:
A type argument has come up before and shouldn't be added because of the following reason in the link you posted:

> `getMeAT<T>(): T`:
If a type parameter does not appear in the types of any parameters, you don't really have a generic function, you just have a disguised type assertion.
Prefer to use a real type assertion, e.g. `getMeAT() as number`.
Example where a type parameter is acceptable: `function id<T>(value: T): T;`.
Example where it is not acceptable: `function parseJson<T>(json: string): T;`.
Exception: `new Map<string, number>()` is OK.


This is because otherwise you'll write code that assumes that all the tokens passed in are well formed which is bad since the tokens are usually passed from other systems or third parties...